RelationalAI/Client.cs

```diff
@@ -477,9 +477,22 @@ public async Task<TransactionAsyncResult> DeleteModelsAsync(string database, str
                 .ExecuteAsync(() => GetTransactionAsync(id));
 
             var transaction = transactionResponse.Transaction;
-            var results = await GetTransactionResultsAsync(id);
-            var metadata = await GetTransactionMetadataAsync(id);
-            var problems = await GetTransactionProblemsAsync(id);
+            List<ArrowRelation> results = null;
+            MetadataInfo metadata = null;
+            List<object> problems = null;
+
+            try
+            {
+                results = await GetTransactionResultsAsync(id);
+                metadata = await GetTransactionMetadataAsync(id);
+                problems = await GetTransactionProblemsAsync(id);
+            }
+            catch (NotFoundException)
+            {
+                // ignore not found exceptions
+                // for aborted transactions results, metadata or problems
+                // might be missing
+            }
```

Comment:
this was the old solution that we used in the Julia SDK, to work around the fact that https://github.com/RelationalAI/raicloud-control-plane/issues/3665 hadn't been solved yet. But now that https://github.com/RelationalAI/raicloud-control-plane/pull/3874 has been merged, we should use that here instead.

Instead of swallowing exceptions like this, which could mask other issues, we think you should change this logic to check for state=ABORTED, and then check the abort_reason from the `transaction` object. The logic should be something like this:
```
List<ArrowRelation> results = null;
MetadataInfo metadata = null;
List<object> problems = null;

if transaction.state == COMPLETED || transaction.abort_reason == "ic violation" {
        results = await GetTransactionResultsAsync(id);
        metadata = await GetTransactionMetadataAsync(id);
        problems = await GetTransactionProblemsAsync(id);
}
```
See more here:
- https://github.com/RelationalAI/rai-sdk-issues/issues/70#issuecomment-1242387662
- https://github.com/RelationalAI/rai-sdk-issues/issues/70#issuecomment-1233564065